hosting/libs/firebase/auth.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  type User,
+  GoogleAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged as _onAuthStateChanged,
+} from "firebase/auth";
+
+import { firebaseAuth } from "./config";
+
+export function onAuthStateChanged(callback: (authUser: User | null) => void) {
+  return _onAuthStateChanged(firebaseAuth, callback);
+}
+
+export async function signInWithGoogle() {
+  const provider = new GoogleAuthProvider();
+
+  try {
+    const result = await signInWithPopup(firebaseAuth, provider);
+
+    if (!result || !result.user) {
+      throw new Error("Google sign in failed");
+    }
+    return result.user.uid;
+  } catch (error) {
+    console.error("Error signing in with Google", error);
+  }
+}
+
+export async function signOutWithGoogle() {
+  try {
+    await firebaseAuth.signOut();
+  } catch (error) {
+    console.error("Error signing out with Google", error);
```

Comment:
It would be better if we could handle this error condition and show some message to user with snackbar

Reply:
Yes, that would be helpful. Just avoid putting the details of the message from the `error` in snackbar. The current console.error is good as it is. But it would indeed be helpful to get a snackbar informing that something went wrong.